File: src/main/java/com/card/loop/xyz/controller/UserController.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.card.loop.xyz.controller;

import com.card.loop.xyz.dto.UserDto;
import com.card.loop.xyz.service.UserService;
import java.net.UnknownHostException;
import java.util.List;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;

import org.springframework.web.bind.annotation.RestController;

/**
 *
 * @author Vine Deiparine
 * @date   09/21/2015
 */
@RestController
@RequestMapping("/user")
public class UserController {
    @Autowired UserService service;
    
    @RequestMapping("/getUser/{user}/{usertype}")
    @ResponseBody
    public UserDto getUser(@PathVariable String user,@PathVariable String usertype) throws UnknownHostException {
        UserDto userinfo = new UserDto();
        if(user.length()>0 && usertype.length()>0)
            userinfo = service.getUserInfo(user,usertype);
        else
            return null;
        return userinfo;
    }

    @RequestMapping("/signup")
    @ResponseBody
    public UserDto signup(@RequestBody UserDto user) throws UnknownHostException
    {
        try{
        boolean ok = service.register(user);
            if(ok==false){
                System.out.println("HAHA");
            }
        }catch(Exception e){
            e.printStackTrace();
        }
        return user;
    }
    
    
    @RequestMapping("/login")
    @ResponseBody
    public UserDto login(@RequestBody UserDto user) throws UnknownHostException
    {
        UserDto result = null;
        try{
            result = service.login(user);
            if(result==null)
                user.getErrorList().add("Username or password is invalid.");
        }catch(Exception e){
            System.out.println(e.toString());
        }
        return result;
    }
    
    //get all new accounts
    @RequestMapping("/newAccountRequests")
    public List<UserDto>  accountRequests(){
        List<UserDto> result = null;
        try{
            result = service.getAllNewAccount();
            if(result==null)
                return null;
        }catch(Exception e){
            e.printStackTrace();
        }
        return result;
    }
    
    @RequestMapping("/developer")
    public List<UserDto>  accountDeveloper(){
        List<UserDto> result = null;
        try{
            result = service.getAllDeveloper();
            if(result==null)
               return null;
        }catch(Exception e){
            e.printStackTrace();
        }
        return result;
    }
    
    @RequestMapping("/validate")
    public boolean validate(UserDto user){return false;}
    
    @RequestMapping("/edit")
    @ResponseBody
    public boolean edit(UserDto user) throws UnknownHostException {
        boolean edited = false;
        if(user!=null) {
            if(user.getPassword()!=null)
                service.editPassword(user);
            else
                service.editEmail(user);
            
            edited = true;
        }
        return edited;
    }
    
    @RequestMapping("/approve")
    public boolean approve(UserDto user) throws UnknownHostException, Exception{
        return false;
    }
    
    @RequestMapping("/promote")
    public boolean promote(UserDto user){return false;}
    
    @RequestMapping("/demote")
    public boolean demote(UserDto user){return false;}
    
    @RequestMapping("/block")
    public List<UserDto> getAllBlockedUsers(){
     List<UserDto> result = null;
        try{
            result = service.getBlockedUsers();
            if(result==null);
             //   user.getErrorList().add("NULLL!");
        }catch(Exception e){
            //user.getErrorList().add(e.toString());
            System.out.println(e.toString());
        }
        return result;
    }
    
   /* @RequestMapping("/blockUser/{id}")
    @ResponseBody
    public boolean blockUser(@RequestBody UserDto user) throws UnknownHostException{
        UserService service = new UserService();
        return service.blockUser(user);
    }*/
   
    @RequestMapping("/blockUser/{username}")
    @ResponseBody
    public boolean blockUser(@PathVariable String username) throws UnknownHostException, Exception{
        UserDto ud = new UserDto();
        ud.setUsername(username);
        return service.block(ud);
    }
    
    @RequestMapping("/unblockUser/{username}")
    @ResponseBody
    public boolean unblockUser(@PathVariable String username) throws UnknownHostException, Exception{
        UserDto ud = new UserDto();
        ud.setUsername(username);
        return service.unblock(ud);
    }
    
    @RequestMapping("/acceptUser/{username}")
    @ResponseBody
    public boolean acceptUser(@PathVariable String username) throws UnknownHostException, Exception{
        UserDto ud = new UserDto();
        ud.setUsername(username);
        return service.accept(ud);
    }
    /**
    @RequestMapping("/changeEmail")
    @ResponseBody
    public boolean changeEmail(@RequestBody UserDto user) throws UnknownHostException, Exception{
        //UserDto ud = new UserDto();
        //ud.setUsername(username);
        //return service.changeEmail(ud);
        UserDto result = null;
        try{
            result = service.changeEmail(user);
            if(result==null)
                user.getErrorList().add("Username or password is invalid.");
        }catch(Exception e){
            System.out.println(e.toString());
        }
        return result;
        
    }**/
    
    @RequestMapping("/reviewer")
    public List<UserDto> reviewerUser(){
     List<UserDto> result = null;
        try{
            result = service.getAllReviewer();
            if(result==null)
                return null;
        }catch(Exception e){
            e.printStackTrace();
        }
        return result;
    }
    
    @RequestMapping("/inactive")
    public List<UserDto> inactiveUser(){
     List<UserDto> result = null;
        try{
            result = service.inactive();
            if(result==null)
                return null;
        }catch(Exception e){
            e.printStackTrace();
        }
        return result;
    }
    
    @RequestMapping("/unblock")
    public boolean unblock(UserDto user){return false;}
    
    @RequestMapping("/admins")
    public List<UserDto> admins(UserDto user){return null;}
    
    @RequestMapping("/commonUsers")
    public List<UserDto> commonUsers(UserDto user){return null;}
}

File: src/main/webapp/WEB-INF/resources/js/scripts/controllers/userController.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */


(function(){
  "use strict";
  
    
app.controller('LoginCtrl', ['$scope', '$store', '$http', 'userService', 'utilService' , function($scope, $store,$http,userService,utilService) {
    $store.bind($scope, 'username', '');
    $store.bind($scope, 'userID', '');
    $store.bind($scope, 'userType', '');
    $store.bind($scope, 'password', '');
    $scope.user = '';
    $scope.userId = '';
    $scope.password = '';
    $scope.type = '';
    $scope.ConfirmPass = '';
    $scope.newEmail = '';
    $scope.currentUser = '';
    $scope.newPass = '';
    $scope.newPass2 = '';

    $http.get("/loop-XYZ/loop/user/getUser/"+$scope.username+"/"+$scope.userType).success(function(response){
           $scope.currentUser = response; 
    });
        
    $scope.$watch("username", function(newVal) {
        console.log(newVal);
    });
    $scope.$watch("userType", function(newVal) {
        console.log(newVal);
    });
    
     $scope.$watch("userID", function(newVal) {
        console.log(newVal);
    });
    $scope.$watch("password", function(newVal) {
        console.log("password"+newVal);
    });
    $scope.hide = true;
    $scope.clearUser = function(){ 
        $store.remove('username'); 
        $store.remove('userID');
        $store.remove('userType');
    };
    
    $scope.ChangeEmail = function(pass,email) {
        if($scope.currentUser.password===pass) {
        var data = {
          "username": $scope.username,
          "email": email,
          "usertype": $scope.userType
        };    
        $http.post("/loop-XYZ/loop/user/edit",data).success(function(response) {
            alert(response);
        });
        } else {
            alert('Failed to change Email! Confirmed Password failed!');
        }
    }
    
    $scope.ChangePass = function(pass,newpass,newpass2) {
        if($scope.currentUser.password===pass) {
            if(newpass===newpass2) {
        var data = {
          "username": $scope.username,
          "password": newpass,
          "usertype": $scope.userType
        };
        $http.post("/loop-XYZ/loop/user/edit",data).success(function(response) {
            alert(response);
        });
            } else {
                alert('Passwords do not match!');
            }
        } else {
            alert('Failed to change Password! Confirmed Password failed!');
        }
    }

    $scope.signup = function(){
        if($scope.newpass===$scope.newpass2) {

         var data =  {
                username: $scope.username,
                password: $scope.newpass,
                email : $scope.email,
                userType :  $('input[name=optionsRadios]:checked','#defaultForm').val()
        }; 
            $http.post('/loop-XYZ/loop/user/signup',data).success(function(data){
                   if(data === null)
                        alert("Registration unSuccessful!");
                   else if(data !== null){
                       if(data === true)
                        alert("Registration Successful!");
                       else
                        alert("User exists!");
                    }
            });
        };

    };
  $scope.login = function(){
        var data =  {
                username: $scope.username,
                password: $scope.password,
                userType :  $('input[name=optionsRadios]:checked','#login_form').val()
        }; 

      userService.login(data).success(function(data){
            $store.bind($scope, 'username', data.username);
            $store.bind($scope, 'userID', data.id);
            $store.bind($scope, 'userType', data.usertype);
                 if(data === null){
                       $('#loginto').prepend("<p class='alert alert-danger'><i class='icon-warning-sign'></i> Invalid account details. Check your information below.</p>");
                   }
                   else if(data.id !== null){
                        if(data.usertype === "developer")
                            window.location = "/loop-XYZ/store/developer-update";
                        else if(data.usertype === "reviewer")
                            window.location = "/loop-XYZ/store/reviewer-update";
                        else 
                            window.location = "/loop-XYZ/store/admin-view";
                   }
        })
      .error(function(response){
           alert("Data Error! :(");
       });
    };
    
   if($store.get('userID').length === 0 && window.location.toString().split('/store/')[1] !== 'home'){
       window.location = '/loop-XYZ/store/home';
   }
}]);


app.controller('developerAccountCtrl', ['$scope', '$http', 'userService', function($scope, $http,userService) {
    userService.getDeveloper()  
    .success(function(data) {
    	$scope.developerAccount = data;
    })
    .error(function(jqXHR, status, error) {
        console.log(""+ error);
    });
}]);

app.controller('reviewerAccountCtrl', ['$scope', '$http', '$rootScope', 'userService', 'utilService', function($scope, $http, $rootScope, userService, utilService) {
    $scope.luckyReviewer = "";
    
    function load() {    
        userService.getReviewer().success(function(response) {
            $scope.reviewerAccount = response;
        })
        .error(function(jqXHR, status, error) {
            console.log(""+ error);
        });
        
        if(utilService.getValue("leid")!==null){
            $http.get("/loop-XYZ/loop/LE/getLE/"+utilService.getValue("leid")).success(function(response) {
                  $scope.les = response;
            });
        }
    }
    
    $scope.assignReviewer = function() {
        if(utilService.getValue("leid")!==null){
            $http.post("/loop-XYZ/loop/LE/assignReviewer?leid="+utilService.getValue("leid")+"&"+"reviewer="+$scope.luckyReviewer).success(function(response) {
                alert(response);
                window.location.reload();
            });
        }
    };    
    
    load();
}]);
app.controller('reviewerAccountCtrlLO', ['$scope', '$http', '$rootScope', 'userService', 'utilService', function($scope, $http, $rootScope, userService, utilService) {
    $scope.luckyReviewer = "";
    
    function load() {    
        userService.getReviewer().success(function(response) {
            $scope.reviewerAccount = response;
        })
        .error(function(jqXHR, status, error) {
            console.log(""+ error);
        });
        
        if(utilService.getValue("loid")!==null){
            $http.get("/loop-XYZ/loop/LO/getLO/"+utilService.getValue("loid")).success(function(response) {
                  $scope.los = response;
            });
        }
    }
    
    $scope.assignReviewer = function() {
        if(utilService.getValue("loid")!==null){
            $http.post("/loop-XYZ/loop/LO/assignReviewer?loid="+utilService.getValue("loid")+"&"+"reviewer="+$scope.luckyReviewer).success(function(response) {
                alert(response);
                window.location.reload();
            });
        }
    };
    
    load();
}]);
//displays all inactive accounts
app.controller('inactiveAccountCtrl', ['$scope', '$http', 'userService', function($scope, $http, userService) {
    userService.getInactive().success(function(data) {
    	$scope.inactiveAccount = data;
    }).error(function(jqXHR, status, error) {
        console.log(""+ error);
    });
}]);

app.controller('reviewerAccountCtrlOldLO', ['$scope', '$http', '$rootScope', 'userService', 'utilService', function($scope, $http, $rootScope, userService, utilService) {
    $scope.luckyReviewer = "";
    
    function load() {    
        userService.getReviewer().success(function(response) {
            $scope.reviewerAccount = response;
        })
        .error(function(jqXHR, status, error) {
            console.log(""+ error);
        });
        
        if(utilService.getValue("loid")!==null){
            $http.get("/loop-XYZ/loop/LO/getOldLO/"+utilService.getValue("loid")).success(function(response) {
                  $scope.oldlos = response;
            });
        }
    }
    
    $scope.assignReviewer = function() {
        if(utilService.getValue("loid")!==null){
            $http.post("/loop-XYZ/loop/LO/assignReviewer?loid="+utilService.getValue("loid")+"&"+"reviewer="+$scope.luckyReviewer).success(function(response) {
                alert(response);
            });
        }
    };
    
    load();
}]);


//displays all blocked accounts
app.controller('blockedAccountCtrl', ['$scope', '$http', 'userService',function($scope, $http, userService) {
    userService.getBlocked().success(function(data) {
    	$scope.blockedAccount = data;
    })
    .error(function(jqXHR, status, error) {
        console.log(""+ error);
    });
}]);

//displays all new accounts
app.controller('newAccountRequestCtrl', ['$scope', '$http', 'userService', function($scope, $http, userService) {
    userService.getNewAccount().success(function(data) {
    	$scope.newAccount = data;
    })
    .error(function(jqXHR, status, error) {
        console.log(""+ error);
    });
}]);

//block an account
 app.controller('blockCtrl', ['$scope', '$http', '$rootScope', 'userService', function($scope, $http, $rootScope, userService) {
         $rootScope.userBlock = '';
   $scope.blockMe = function(){ 
        userService.blockUser($rootScope.userBlock).success(function(data) {
            window.location.href="admin-view4";
        })
        .error(function(jqXHR, status, error) {
            console.log("BLOCK USER OPERATTION ERROR : "+ error);
        });
   };
   $scope.unblockMe = function(){ 
        userService.unblockUser($rootScope.userBlock).success(function(data) {
            window.location.reload(true);
        })
        .error(function(jqXHR, status, error) {
            console.log("UNBLOCK USER OPERATTION ERROR : "+ error);
        });
   };
   
   $scope.assignUser = function(developer) {
       $rootScope.userBlock = developer.username;
   };
}]);

//accept an account
app.controller('acceptCtrl', ['$scope', '$http', '$rootScope', 'userService', function($scope, $http, $rootScope, userService) {
    $rootScope.userAccept = '';
    $scope.acceptMe = function(){ 
        userService.acceptUser($rootScope.userAccept).success(function(data) {
            window.location.reload(true);
        }).error(function(jqXHR, status, error) {
            console.log("ACCEPT USER ERROR: "+ error);
        });
   };
    
   $scope.assignUser = function(developer) {
       $rootScope.userAccept = developer.username;
   };
}]);

})();